Error building non-portable against OpenSSL 1.1

With OpenSSL 1.1 development headers installed, I can't build corefx in non-portable mode:

    $ git checkout master 
    Already on 'master'
    Your branch is up to date with 'upstream/master'.
    $ git rev-parse --short HEAD
    758c059b3d
    $ git clean -xdf
    Removing artifacts/
    $ cd src/Native/
    $ ./build-native.sh

```
home/omajid/devel/dotnet/corefx/src/Native/Unix/System.Security.Cryptography.Native/openssl.c:432:12: error:
      'sk_ASN1_OBJECT_num' was marked unused but was used [-Werror,-Wused-but-marked-unused]
    return sk_ASN1_OBJECT_num(eku);
           ^
/home/omajid/devel/dotnet/corefx/src/Native/Unix/System.Security.Cryptography.Native/openssl.c:448:12: error:
      'sk_ASN1_OBJECT_value' was marked unused but was used [-Werror,-Wused-but-marked-unused]
    return sk_ASN1_OBJECT_value(eku, loc);
           ^
/home/omajid/devel/dotnet/corefx/src/Native/Unix/System.Security.Cryptography.Native/openssl.c:599:29: error:
      'sk_GENERAL_NAME_num' was marked unused but was used [-Werror,-Wused-but-marked-unused]
            for (i = 0; i < sk_GENERAL_NAME_num(altNames); ++i)
                            ^
/home/omajid/devel/dotnet/corefx/src/Native/Unix/System.Security.Cryptography.Native/openssl.c:601:41: error:
      'sk_GENERAL_NAME_value' was marked unused but was used [-Werror,-Wused-but-marked-unused]
                GENERAL_NAME* altName = sk_GENERAL_NAME_value(altNames, i);
                                        ^
/home/omajid/devel/dotnet/corefx/src/Native/Unix/System.Security.Cryptography.Native/openssl.c:646:25: error:
      'sk_GENERAL_NAME_free' was marked unused but was used [-Werror,-Wused-but-marked-unused]
                        sk_GENERAL_NAME_free(altNames);
                        ^
/home/omajid/devel/dotnet/corefx/src/Native/Unix/System.Security.Cryptography.Native/openssl.c:652:13: error:
      'sk_GENERAL_NAME_free' was marked unused but was used [-Werror,-Wused-but-marked-unused]
            sk_GENERAL_NAME_free(altNames);
            ^
/home/omajid/devel/dotnet/corefx/src/Native/Unix/System.Security.Cryptography.Native/openssl.c:859:21: error:
      'sk_GENERAL_NAME_num' was marked unused but was used [-Werror,-Wused-but-marked-unused]
        int count = sk_GENERAL_NAME_num(san);
                    ^
/home/omajid/devel/dotnet/corefx/src/Native/Unix/System.Security.Cryptography.Native/openssl.c:863:38: error:
      'sk_GENERAL_NAME_value' was marked unused but was used [-Werror,-Wused-but-marked-unused]
            GENERAL_NAME* sanEntry = sk_GENERAL_NAME_value(san, i);
                                     ^
/home/omajid/devel/dotnet/corefx/src/Native/Unix/System.Security.Cryptography.Native/openssl.c:943:21: error:
      'sk_GENERAL_NAME_num' was marked unused but was used [-Werror,-Wused-but-marked-unused]
        int count = sk_GENERAL_NAME_num(san);                                                                                             ^
/home/omajid/devel/dotnet/corefx/src/Native/Unix/System.Security.Cryptography.Native/openssl.c:947:38: error:
      'sk_GENERAL_NAME_value' was marked unused but was used [-Werror,-Wused-but-marked-unused]
            GENERAL_NAME* sanEntry = sk_GENERAL_NAME_value(san, i);
                                     ^
/home/omajid/devel/dotnet/corefx/src/Native/Unix/System.Security.Cryptography.Native/openssl.c:1011:12: error:
      'sk_X509_num' was marked unused but was used [-Werror,-Wused-but-marked-unused]
    return sk_X509_num(stack);
           ^
/home/omajid/devel/dotnet/corefx/src/Native/Unix/System.Security.Cryptography.Native/openssl.c:1027:12: error:
      'sk_X509_value' was marked unused but was used [-Werror,-Wused-but-marked-unused]
    return sk_X509_value(stack, loc);
           ^
/home/omajid/devel/dotnet/corefx/src/Native/Unix/System.Security.Cryptography.Native/openssl.c:1039:5: error:
      'sk_X509_pop_free' was marked unused but was used [-Werror,-Wused-but-marked-unused]
    sk_X509_pop_free(stack, X509_free);
    ^
/home/omajid/devel/dotnet/corefx/src/Native/Unix/System.Security.Cryptography.Native/openssl.c:1197:12: error:
      'sk_X509_new_null' was marked unused but was used [-Werror,-Wused-but-marked-unused]
    return sk_X509_new_null();
           ^
/home/omajid/devel/dotnet/corefx/src/Native/Unix/System.Security.Cryptography.Native/openssl.c:1218:12: error:
      'sk_X509_push' was marked unused but was used [-Werror,-Wused-but-marked-unused]
    return sk_X509_push(stack, x509);
           ^
15 errors generated.
```

These functions are defined in [safestack.h](https://github.com/openssl/openssl/blob/master/include/openssl/safestack.h) along the lines of:

    static ossl_unused ossl_inline int sk_##t1##_num(const STACK_OF(t1) *sk) 

And `ossl_unused` is defined in [e_os2.h](https://github.com/openssl/openssl/blob/7120f1e7c74c8220f43fd28ac7303c0cab29dabe/include/openssl/e_os2.h#L290) as:

```
/* ossl_unused: portable unused attribute for use in public headers */
# if defined(__GNUC__)
#  define ossl_unused __attribute__((unused))
# else
#  define ossl_unused
# endif
```

--- a/src/Native/Unix/System.Security.Cryptography.Native/CMakeLists.txt
+++ b/src/Native/Unix/System.Security.Cryptography.Native/CMakeLists.txt
@@ -6,6 +6,7 @@ set(CMAKE_INCLUDE_CURRENT_DIR ON)
 add_compile_options(-Wno-cast-align)
 add_compile_options(-Wno-reserved-id-macro)
 add_compile_options(-Wno-documentation)
+add_compile_options(-Wno-used-but-marked-unused)
 
 add_definitions(-DPIC=1 -DOPENSSL_API_COMPAT=0x10100000L)
 

